psydac/core/bsplines_kernels.py

```diff
@@ -1087,13 +1087,19 @@ def cell_index_p(breaks: 'float[:]', i_grid: 'float[:]', tol: float, out: 'int[:
         # Binary search
         low, high = 0, nbk - 1
         i_cell = (low + high)//2
-        while  x < breaks[i_cell] - tol or x >= breaks[i_cell + 1] + tol:
+        it = 0
+        max_it = 2*(high-low) # this number of iterations should not be reached
+        while it < max_it and (x < breaks[i_cell] - tol or x >= breaks[i_cell + 1] + tol):
             if x < breaks[i_cell]:
                 high = i_cell
             else:
                 low = i_cell
             i_cell = (low + high)//2
-
+            it += 1
+        if it >= max_it:
+            print("Warning: binary search loop is too long in cell_index_p -- aborted")
```

Comment:
We already raise a `ValueError` in the calling function, hence I see limited value in this print statement. I suggest to remove it.

Reply:
Indeed, I wrote it before changing the return value, now it is useless.